app/scripts/models/auth_brokers/oauth.js

```diff
@@ -111,7 +90,21 @@ define(function (require, exports, module) {
           }
           return this._oAuthClient.getCode(oauthParams);
         })
-        .then(_formatOAuthResult);
+        .then((result) => {
+          if (! result) {
+            return Promise.reject(OAuthErrors.toError('INVALID_RESULT'));
+          }
+          // The oauth-server would previously construct and return the full redirect URI,
+          // but we now expect to receive `code` and `state` and build it ourselves
+          // using the relier's locally-validated redirectUri.
```

Comment:
I just had a look at where [redirectUri comes into the app](https://github.com/mozilla/fxa-content-server/blob/fc2a86d86d66541d6037b22d5ba61f41cceede7b/app/scripts/models/reliers/oauth.js#L24), and the new approach does seem slightly safer.